2001-03-16  Jakub Jelinek  <jakub@redhat.com>

	* crtstuff.c (init_dummy): Remove obsolete i386 linux crtendS.o
	kludge.

--- gcc/crtstuff.c.jj	Thu Mar 15 10:55:50 2001
+++ gcc/crtstuff.c	Fri Mar 16 13:00:10 2001
@@ -414,21 +414,6 @@ init_dummy (void)
   FORCE_INIT_SECTION_ALIGN;
 #endif
   asm (TEXT_SECTION_ASM_OP);
-
-/* This is a kludge. The i386 GNU/Linux dynamic linker needs ___brk_addr,
-   __environ and atexit (). We have to make sure they are in the .dynsym
-   section. We accomplish it by making a dummy call here. This
-   code is never reached.  */
- 
-#if defined(__linux__) && defined(__PIC__) && defined(__i386__)
-  {
-    extern void *___brk_addr;
-    extern char **__environ;
-
-    ___brk_addr = __environ;
-    atexit (0);
-  }
-#endif
 }
 
 #else  /* OBJECT_FORMAT_ELF */
